### PR TITLE
rust 1.71.1

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by chenrui333** [comment]
🏃

File: Formula/rust.rb [hunk 1]
```diff
@@ -4,14 +4,15 @@ class Rust < Formula
...
+    url "https://static.rust-lang.org/dist/rustc-1.71.1-src.tar.gz"
+    sha256 "6fa90d50d1d529a75f6cc349784de57d7ec0ba2419b09bde7d335c25bd4e472e"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
       url "https://github.com/rust-lang/cargo.git",
-          tag:      "0.72.0",
-          revision: "cfd3bbd8fe4fd92074dfad04b7eb9a923646839f"
+          # TODO: uncomment when tagged
+          # tag:      "0.72.2",
```
Comment (chenrui333) [inline Formula/rust.rb:14]: did we raise an upstream issue for it? (I did not find any)

Comment (chenrui333) [reply Formula/rust.rb:14]: just raised an issue on this, https://github.com/rust-lang/cargo/issues/12460

Comment (chenrui333) [reply Formula/rust.rb:14]: now we have the tag.

File: Formula/rust.rb [hunk 2]
```diff
@@ -86,7 +86,7 @@ def install
       ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
     end
 
-    args = %W[--prefix=#{prefix} --enable-vendor --set rust.jemalloc]
+    args = %W[--prefix=#{prefix} --sysconfdir=#{etc} --enable-vendor --set rust.jemalloc]
```
Comment (ZhongRuoyu) [inline Formula/rust.rb:89]: This is a surprisingly old issue: https://github.com/rust-lang/cargo/issues/4070

I wonder why we didn't have this problem before.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
no space on linux build

```
Caused by:
    0: failed to copy '/tmp/rust-20230816-8015-zlpjgn/rustc-1.71.1-src/build/tmp/tarball/llvm-tools/x86_64-unknown-linux-gnu/image/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc' to '/tmp/rust-20230816-8015-zlpjgn/rustc-1.71.1-src/build/tmp/tarball/llvm-tools/x86_64-unknown-linux-gnu/llvm-tools-1.71.1-x86_64-unknown-linux-gnu/llvm-tools-preview/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc'
    1: No space left on device (os error 28)
```

rerun with self-hosted runner.

**Review by iMichka — APPROVED** [review]

**Comment by chenrui333** [comment]
should be good to go, previous run, https://github.com/Homebrew/homebrew-core/actions/runs/5882654639/attempts/1?pr=139719

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**